### PR TITLE
Perf: use StringSet instead for better lookup performance

### PR DESCRIPTION
**Pull request by chuganzy**
`isNameTooSimple` function in `lib/DriverTool/swift_api_digester_main.cpp` had a room to slightly improve the performance.

This PR updates `SimpleNames` to be `StringSet` in place of `vector` to make its time complexity to `O(1)` from `O(n)`.

### CODE REVIEW

**Comment by tschuett** [comment]
My only comment is: LLVM never uses `unordered_set`.
https://llvm.org/docs/ProgrammersManual.html#other-set-like-container-options
How about a `StringSet`:
https://llvm.org/docs/ProgrammersManual.html#llvm-adt-stringset-h

**Comment by chuganzy** [comment]
@tschuett Thank you for the comment, that's good to know! Especially since in this function we only need `StringRef` check, `StringSet` seems to be the perfect match!

File: lib/DriverTool/swift_api_digester_main.cpp [hunk 1]
```diff
@@ -273,11 +273,10 @@ class RemovedAddedNodeMatcher : public NodeMatcher, public MatchedNodeListener {
   }
 
   static bool isNameTooSimple(StringRef N) {
-    static std::vector<std::string> SimpleNames = {"unit", "data", "log", "coding",
+    static llvm::StringSet<> SimpleNames = {"unit", "data", "log", "coding",
       "url", "name", "date", "datecomponents", "notification", "urlrequest",
       "personnamecomponents", "measurement", "dateinterval", "indexset"};
-    return std::find(SimpleNames.begin(), SimpleNames.end(), N) !=
-      SimpleNames.end();
+    return SimpleNames.contains(N);
```
Comment (CodaFi) [inline lib/DriverTool/swift_api_digester_main.cpp:279]: Do you have benchmarks to show that this is a demonstrable win? Often, linear search across small, fixed datasets like the above can be much faster than the memops required to execute a hash chain walk.

Comment (chuganzy) [reply lib/DriverTool/swift_api_digester_main.cpp:279]: Wow good to learn! Let me do a quick benchmark and get back to you (probably later this week!)

Comment (chuganzy) [reply lib/DriverTool/swift_api_digester_main.cpp:279]: Sorry for being late here. I made a quick benchmark project to compare the actual performance. The code is as simple as bellow. Please let me know if there is anything wrong, but seems like the change in this PR makes it more than 4x faster.

Screen Shot 2022-06-24 at 1 13 47 PM

```cpp
#include <llvm/ADT/StringSet.h>
#include <llvm/ADT/StringRef.h>
#include <benchmark/benchmark.h>

static bool isNameTooSimpleOriginal(llvm::StringRef N) {
    static std::vector<std::string> SimpleNames = {"unit", "data", "log", "coding",
        "url", "name", "date", "datecomponents", "notification", "urlrequest",
        "personnamecomponents", "measurement", "dateinterval", "indexset"};
    return std::find(SimpleNames.begin(), SimpleNames.end(), N) !=
      SimpleNames.end();
}

static bool isNameTooSimpleNew(llvm::StringRef N) {
    static llvm::StringSet<> SimpleNames = {"unit", "data", "log", "coding",
        "url", "name", "date", "datecomponents", "notification", "urlrequest",
        "personnamecomponents", "measurement", "dateinterval", "indexset"};
    return SimpleNames.contains(N);
}

static void Original(benchmark::State& state) {
  for (auto _ : state) {
    isNameTooSimpleOriginal("datecomponents");
    isNameTooSimpleOriginal("dummy");
  }
}

BENCHMARK(Original);

static void New(benchmark::State& state) {
  for (auto _ : state) {
    isNameTooSimpleNew("dummy");
    isNameTooSimpleNew("datecomponents");
  }
}

BENCHMARK(New);
```

Comment (chuganzy) [reply lib/DriverTool/swift_api_digester_main.cpp:279]: Oh wait.. If I do a release build, I got a worse result..
Screen Shot 2022-06-24 at 2 39 34 PM

**Comment by chuganzy** [comment]
Let me close this for this reason: https://github.com/apple/swift/pull/42284/files#r906426233
Thank you all for the inputs!

**Pull request closed without merging**